docker_test.go

```diff
@@ -1382,6 +1383,26 @@ func TestContainerNonExistentImage(t *testing.T) {
 	})
 }
 
+func TestContainerCustomPlatformImage(t *testing.T) {
+	t.Run("Use a custom image platform", func(t *testing.T) {
+		c, err := GenericContainer(context.Background(), GenericContainerRequest{
+			ContainerRequest: ContainerRequest{
+				Image:         "redis:latest",
+				SkipReaper:    true,
+				ImagePlatform: "windows/arm12",
+			},
+			Started: false,
+		})
+		if c != nil {
+			defer c.Terminate(context.Background())
+		}
+		if err == nil {
+			t.Fatalf("Expected non-nil error")
```

Comment:
Because the test expects to raise an error, I would log it with more information:
```suggestion
			t.Fatalf("Expected non-nil error when using an invalid platform: 'windows/arm12'")
```
Maybe extracting the platform to a variable to be reused in this message? wdyt?